examples/react/next-app-router/app/page.tsx

```diff
@@ -0,0 +1,50 @@
+'use client';
+
+import algoliasearch from 'algoliasearch/lite';
+import { Hit as AlgoliaHit } from 'instantsearch.js';
+import { ServerInsertedHTMLContext } from 'next/navigation';
+import React from 'react';
+import {
+  InstantSearch,
+  Hits,
+  Highlight,
+  SearchBox,
+  RefinementList,
+  InstantSearchWrapper,
+} from 'react-instantsearch';
+
+const client = algoliasearch('latency', '6be0576ff61c053d5f9a3225e2a90f76');
+
+type HitProps = {
+  hit: AlgoliaHit<{
+    name: string;
+    price: number;
+  }>;
+};
+
+function Hit({ hit }: HitProps) {
+  return (
+    <>
+      <Highlight hit={hit} attribute="name" className="Hit-label" />
+      <span className="Hit-price">${hit.price}</span>
+    </>
+  );
+}
+
+export default function SearchPage() {
+  return (
+    <InstantSearchWrapper ServerInsertedHTMLContext={ServerInsertedHTMLContext}>
+      <InstantSearch searchClient={client} indexName="instant_search">
```

Comment:
Might simply be `<NextInstantSearch searchClient={client} indexName="instant_search">` if we package it separately.

Reply:
Some of it does, as injecting `initialResults` uses a Next.js-specific API.
Most of the changes will be directly in `react-instantsearch`, with tiny wrappers for frameworks that just inject initial results with their framework's API.

Reply:
Or they could indeed just pass their own `insertHTML` function, but I think it's nice to just have to replace `<InstantSearch>` with no prop change